interfaces/policy/policy.go

```diff
@@ -116,22 +116,23 @@ func (ic *InstallCandidate) Check() error {
 
 // ConnectCandidate represents a candidate connection.
 type ConnectCandidate struct {
-	// TODO: later we need to carry dynamic attributes once we have those
 	Plug                *snap.PlugInfo
 	PlugSnapDeclaration *asserts.SnapDeclaration
+	PlugAttrs           map[string]interface{}
 
 	Slot                *snap.SlotInfo
 	SlotSnapDeclaration *asserts.SnapDeclaration
+	SlotAttrs           map[string]interface{}
 
 	BaseDeclaration *asserts.BaseDeclaration
 }
 
 func (connc *ConnectCandidate) plugAttrs() map[string]interface{} {
-	return connc.Plug.Attrs
+	return connc.PlugAttrs
```

Comment:
as I said in the standup with the this change, while we'll need to test the new paths, we can undo the changes to many tests, at least most in policy itself and basedeclaration